### PR TITLE
fix doc errors

### PR DESCRIPTION
**Pull request by yannikschaelte**
* Fix documentation build errors.
* Test create docs on pull requests.
* Add favicon for docs.

Readthedocs complained about these things. On local build, I now have no more errors.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,11 +20,12 @@ addons:
       - libatlas-base-dev
       - lcov
       - swig3.0
+      - pandoc
 
 install:
   - mkdir -p ~/bin/ && ln -s /usr/bin/swig3.0 ~/bin/swig && export PATH=~/bin/:$PATH
   - pip3 install --upgrade setuptools wheel pkgconfig
-  - pip3 install --upgrade jupyter # for jupyter notebooks
+  - pip3 install --upgrade jupyter
```
Comment (dweindl) [inline .travis.yml]: Why not put all of them into `travis_pip_reqs`?

File: .travis.yml [hunk 2]
```diff
@@ -33,8 +34,9 @@ script:
   - python3 -m flake8 --exclude=build,doc,example,tmp,amici_models
   - travis_wait 20 python3 -m pytest --cov=pypesto ./test/test_*
   - travis_wait 20 xvfb-run -a python3 -m pytest --cov=pypesto --cov-append ./test/visualize/test_*
-  - coverage xml
+  - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then sphinx-build -W -b html -d doc/_build/html doc/ doc/_build/html; fi
```
Comment (dweindl) [inline .travis.yml]: Can that go into the sphinx Makefile?

Comment (yannikschaelte) [reply .travis.yml]: why?

Comment (dweindl) [reply .travis.yml]: because `make html` is way more readable

Comment (yannikschaelte) [reply .travis.yml]: yeah but I would leave the default options in make, s.t. the docs do not error upon minor inconsistencies (which are turned into errors here via -W). but feel free to edit it, possibly with options.

Comment (dweindl) [reply .travis.yml]: `make travis` `make rtd` ... ?

**Pull request merged**